### DISCUSSION TITLE
Snap to 2 custom zones?

### DISCUSSION THREAD

**Discussion by jrwagz** [rxhanson/RectanglePro-Community#491] [Ideas]
I'm coming from having used FancyZones for a few years now, and one of the features from that which I really like, is the ability to snap to one or two zones.  To snap to two zones, you simple hover over the boundary of the two zones, and it presents the option to snap to both zones at the same time.

See [docs](https://learn.microsoft.com/en-us/windows/powertoys/fancyzones#snapping-to-two-zones-by-hovering-the-edges) for details.

There are also additional options in FancyZones for snapping to multiple zones, see https://learn.microsoft.com/en-us/windows/powertoys/fancyzones#snapping-to-multiple-zones.  Perhaps any of those options to enable this would be useful.

So, in Rectangle Pro, I'd like to have this work as follows:

- be able to define multiple custom zones (non-overlapping)
- enable them all as snap targets
- when snap targets is enabled, and you are choosing which to snap to, if you go to the. boundary of any two of those zones, then it presents the option to snap into the area defined as the sum of the two zones.

Is this possible?

As an attempted workaround, I tried defining another custom zone as the sum of two other zones, and enabled them all for snap areas.  What I found is that only the first zone listed in the custom zone list would be presented as an option, and not any other ones which were in the same area.  That makes sense, and I think it should behave that way.  

I'd just really like the option to be able to snap to two custom zones at once, or just one zone.

Hopefully this makes sense, if not, then I could probably do a screen recording in FancyZones that explains this behavior more clearly.

**Comment by rxhanson** [comment]
Thanks for the feedback. It makes sense - I've watched a video about FancyZones before that demonstrates how this works, but I'll have to install windows at some point to get a better feel for it. I had been putting this off until I could get around to building out more of a UI that can support and communicate this. I'll post back here when I get a chance to work through it.
